allen-cell-animated/timelapse-colorizer
# feature: Sync vector state to URL

Problem
=======
Closes #471, "sync vector state to URL."

*Estimated review size: small, 15 minutes*

Solution
========
- Adds handlers for serializing and deserializing vector URL state.
- Updates names for some decoding methods to simplify `url_utils` parsing.

## Type of change
* New feature (non-breaking change which adds functionality)


Steps to Verify:
----------------
1. Open preview link: https://allen-cell-animated.github.io/timelapse-colorizer/pr-preview/pr-488/viewer?collection=https%3A%2F%2Fallencell.s3.amazonaws.com%2Faics%2Fnuc-morph-dataset%2Ftimelapse_feature_explorer_datasets%2Flineage-annotated_dataset%2Fcollection.json&dataset=Small&feature=volume&t=251&vc=1&vc-color=ff0000&vc-key=_motion_&vc-scale=18&vc-time-int=12&vc-tooltip=c
2. You should see the same settings as below: 

![image](https://github.com/user-attachments/assets/1945ec74-8c16-49f7-8722-fdb737bf8dc1)
